File: ForumFrontend/forumfrontend/src/interfaces/Post.ts
export interface Post {
	id: number;
	title: string;
	description: string;
	userId: string;
	author: string;
	dateOfCreation: Date;
	numberOfComments: number;
	categoryId: number;
}

export type CreatePostData = {
	title: string;
	description: string;
	//userId: string;
	categoryId: number;
	//dateOfCreation: string;
};

File: ForumBackend/ForumBackend/Models/Post.cs
using System.ComponentModel.DataAnnotations;

namespace ForumBackend.Models
{
    public class Post
    {
        public int Id { get; set; }
        [Required]
        [StringLength(300)]
        [MinLength(1)]
        public string Title { get; set; } = string.Empty;
        [Required]
        [MinLength(1)]
        public string Description { get; set; } = string.Empty;
        public DateTime DateOfCreation { get; set; } = DateTime.Now;
        public List<Comment> Comments { get; set; } = new List<Comment>();

        public int CategoryId { get; set; }
        public Category Category { get; set; } = null!;

        public string UserId { get; set; } = string.Empty;
        public User User { get; set; } = null!;
    }
}
